core/src/hlist.rs

```diff
@@ -1542,6 +1576,36 @@ mod tests {
         assert_eq!(folded, 9001)
     }
 
+    #[test]
+    fn test_poly_foldr_consuming() {
+        trait Dummy {
+            fn dummy(&self) -> i32 {
+                1
+            }
+        }
+        impl<T: ?Sized> Dummy for T {}
```

Comment:
Is this `?Sized` a leftover from some earlier version that made use of the bound?  (or is it just a "if you can, you might as well" sort of thing)

Reply:
It's "I have a habit of unconstraining generic trait impls whenever I write them"; mostly due to the first time I forgot to write `T: ?Sized` and got messy compiler errors.